### PR TITLE
[NetAppFiles] Anf 19190 update for 2022 01 01

### PR DESCRIPTION
**Pull request by audunn**
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i>

### Changelog
Add a changelog entry for this PR by answering the following questions:
  1. What's the purpose of the update?
      - [ ] new service onboarding
      - [x] new API version
      - [ ] update existing version for new feature
      - [x] update existing version to fix swagger quality issue in s360
      - [ ] Other, please clarify
  2. When are you targeting to deploy the new service/feature to public regions? Please provide the date or, if the date is not yet available, the month.
  3. When do you expect to publish the swagger? Please provide date or, the the date is not yet available, the month.
  4. If updating an existing version, please select the specific language SDKs and CLIs that must be refreshed after the swagger is published.
      - [ ] SDK of .NET (need service team to ensure code readiness)
      - [ ] SDK of Python
      - [ ] SDK of Java
      - [ ] SDK of Js
      - [ ] SDK of Go
      - [ ] PowerShell
      - [ ] CLI
      - [ ] Terraform
      - [ ] No refresh required for updates in this PR

### Contribution checklist:
- [x] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of "no breaking changes"
- [x] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [x] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist 

> **Applicability**: :warning: 
>
> If your changes encompass only the following scenarios, you should SKIP this section, as these scenarios do not require ARM review.
> - Change to data plane APIs 
> - Adding new properties 
> - All removals

Otherwise your PR may be subject to ARM review requirements. Complete the following:
- [x] Check this box if any of the following apply to the PR so that label "WaitForARMFeedback" will be added automatically to begin ARM API Review. Failure to comply may result in delays to the manifest.
  - Adding a new service
  - Adding new API(s)
  - Adding a new API version
    -[ ] To review changes efficiently, ensure you are using OpenAPIHub to initialize the PR for adding a new version. More details, refer to the [wiki](https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/208/OpenAPI-Hub-Adding-new-API-version).

- [x] Ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If any of the following scenarios apply to the PR, request approval from the Breaking Change Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 
- [ ] Removing API(s) in a stable version
- [ ] Removing properties in a stable version
- [ ] Removing API version(s) in a stable version
- [ ] Updating API in a stable or public preview version with Breaking Change Validation errors
- [ ] Updating API(s) in public preview over 1 year (refer to [Retirement of Previews](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37683/Retirement-of-Previews))

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Addition details on the process and office hours are on the [Breaking change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

**Comment by rkmanda** [comment]
>   "host": "management.azure.com",

pl address the breaking changes before ARM can review it

---
Refers to: specification/netapp/resource-manager/Microsoft.NetApp/stable/2022-01-01/netapp.json:20 in df0ce43. [](commit_id = df0ce43cebad9f331312d706d23878c7223efcef, deletion_comment = False)

**Review by rkmanda — CHANGES_REQUESTED** [review]
:clock1:

**Comment by audunn** [comment]
Breaking change review issue https://msazure.visualstudio.com/One/_workitems/edit/14010855

**Comment by audunn** [comment]
Note we are updating older api version to add `systemData`  that is being flagged by s360  live validation.
https://portal.azure-devex-tools.com/amekpis/correctness/detail?errorId=3D278494-C96B-425C-A6E9-7B414CD640ED

**Comment by audunn** [comment]
breaking change review in progress

**Comment by audunn** [comment]
> > "host": "management.azure.com",
> 
> pl address the breaking changes before ARM can review it
> 
> Refers to: specification/netapp/resource-manager/Microsoft.NetApp/stable/2022-01-01/netapp.json:20 in [df0ce43](https://github.com/Azure/azure-rest-api-specs/commit/df0ce43cebad9f331312d706d23878c7223efcef). [](commit_id = [df0ce43](https://github.com/Azure/azure-rest-api-specs/commit/df0ce43cebad9f331312d706d23878c7223efcef), deletion_comment = False)

@rkmanda Breaking change review has been approved can we move this along now?

**Comment by audunn** [comment]
@rkmanda breaking changes approved
> 🕐

File: specification/netapp/resource-manager/Microsoft.NetApp/stable/2021-08-01/netapp.json [hunk 1]
```diff
@@ -5004,6 +5012,11 @@
           "description": "Snapshot policy Properties",
           "$ref": "#/definitions/snapshotPolicyProperties",
           "x-ms-client-flatten": true
+        },
+        "systemData": {
+          "readOnly": true,
+          "description": "The system meta data relating to this resource.",
+          "$ref": "../../../../../common-types/resource-management/v2/types.json#/definitions/systemData"
         }
```
Comment (raosuhas) [inline specification/netapp/resource-manager/Microsoft.NetApp/stable/2021-08-01/netapp.json:5020]: consider using the commontypes [https://github.com/Azure/azure-rest-api-specs/blob/main/specification/common-types/resource-management/v3/types.json](https://github.com/Azure/azure-rest-api-specs/blob/main/specification/common-types/resource-management/v3/types.json)

**Comment by raosuhas** [comment]
>           "NetApp Resource"

needs to be a POST since it is an action taken on a resource

---
Refers to: specification/netapp/resource-manager/Microsoft.NetApp/stable/2022-01-01/netapp.json:300 in 7fb168c. [](commit_id = 7fb168ce53cf050d217994c22af61eec411eb001, deletion_comment = False)

**Comment by raosuhas** [comment]
>             "description": "Accepted -- Create request accepted; operation will complete asynchronously"

For PUT operations : 

[https://github.com/Azure/azure-resource-manager-rpc/blob/2e1f2c11ce4cfdf309f9cc8ae71575cb8fb83319/v1.0/resource-api-reference.md#put-resource](https://github.com/Azure/azure-resource-manager-rpc/blob/2e1f2c11ce4cfdf309f9cc8ae71575cb8fb83319/v1.0/resource-api-reference.md#put-resource)

The resource provider should return 200 (OK) or 201 (Created) to indicate that the operation completed successfully synchronously.

If the create request cannot be fulfilled quickly, the RP should return and follow the Asynchronous Operations addendum. The resource provider can return 202 (Accepted) in these cases, but 201 + provisioningState is generally preferred.



Generally we also expect to return 200 in case of re-PUT 

---
Refers to: specification/netapp/resource-manager/Microsoft.NetApp/stable/2022-01-01/netapp.json:3337 in 7fb168c. [](commit_id = 7fb168ce53cf050d217994c22af61eec411eb001, deletion_comment = False)

**Comment by raosuhas** [comment]
>             "description": "Accepted -- Create request accepted; operation will complete asynchronously"

Also the 202 is not generally preferred as explained here : 

[https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/async-api-reference.md#creatingupdating-using-put](https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/async-api-reference.md#creatingupdating-using-put)

---
In reply to: [1113872226](https://github.com/Azure/azure-rest-api-specs/pull/18516#issuecomment-1113872226) [](http://example.com/codeflow?ancestors=1113872226)

---
Refers to: specification/netapp/resource-manager/Microsoft.NetApp/stable/2022-01-01/netapp.json:3337 in fba45d9. [](commit_id = fba45d9737ca848858bab7dc5a58e949b115ceab, deletion_comment = False)

**Comment by raosuhas** [comment]
>             "description": "Accepted -- Delete request accepted; operation will complete asynchronously"

do you support both long running and sync modes ?

---
Refers to: specification/netapp/resource-manager/Microsoft.NetApp/stable/2022-01-01/netapp.json:3451 in 7fb168c. [](commit_id = 7fb168ce53cf050d217994c22af61eec411eb001, deletion_comment = False)

**Comment by raosuhas** [comment]
>                 "example": true

conside rmaking this an enum : [https://armwiki.azurewebsites.net/rp_onboarding/process/api_review_best_practices.html#common-issues-found-in-review](https://armwiki.azurewebsites.net/rp_onboarding/process/api_review_best_practices.html#common-issues-found-in-review)

---
Refers to: specification/netapp/resource-manager/Microsoft.NetApp/stable/2022-01-01/netapp.json:4322 in 7fb168c. [](commit_id = 7fb168ce53cf050d217994c22af61eec411eb001, deletion_comment = False)

**Comment by raosuhas** [comment]
>         },

consider using enum as explained above

---
Refers to: specification/netapp/resource-manager/Microsoft.NetApp/stable/2022-01-01/netapp.json:5557 in 7fb168c. [](commit_id = 7fb168ce53cf050d217994c22af61eec411eb001, deletion_comment = False)

**Comment by raosuhas** [comment]
>         "tags": {

consider using common types

---
Refers to: specification/netapp/resource-manager/Microsoft.NetApp/stable/2022-01-01/netapp.json:7167 in 7fb168c. [](commit_id = 7fb168ce53cf050d217994c22af61eec411eb001, deletion_comment = False)

**Comment by raosuhas** [comment]
>         "provisioningState": {

enum or use from commontypes

---
Refers to: specification/netapp/resource-manager/Microsoft.NetApp/stable/2022-01-01/netapp.json:7187 in 7fb168c. [](commit_id = 7fb168ce53cf050d217994c22af61eec411eb001, deletion_comment = False)

**Comment by raosuhas** [comment]
> @rkmanda breaking changes approved
> 
> > 🕐

Added some comments on the latest version.

**Comment by audunn** [comment]
> > ```
> >       "NetApp Resource"
> > ```
> 
> needs to be a POST since it is an action taken on a resource
> 
> Refers to: specification/netapp/resource-manager/Microsoft.NetApp/stable/2022-01-01/netapp.json:300 in [7fb168c](https://github.com/Azure/azure-rest-api-specs/commit/7fb168ce53cf050d217994c22af61eec411eb001). [](commit_id = [7fb168c](https://github.com/Azure/azure-rest-api-specs/commit/7fb168ce53cf050d217994c22af61eec411eb001), deletion_comment = False)

@raosuhas we are moving this from the preview api-version to stable. This was approved here, we are moving  https://github.com/Azure/azure-rest-api-specs/pull/14286

File: specification/netapp/resource-manager/Microsoft.NetApp/stable/2021-08-01/netapp.json (hunk 1 above)
Comment (audunn) [reply specification/netapp/resource-manager/Microsoft.NetApp/stable/2021-08-01/netapp.json:5020]: sorry I don't get your point, `systemData` is being referenced from common types.

**Comment by audunn** [comment]
> > ```
> >     "tags": {
> > ```
> 
> consider using common types
> 
> Refers to: specification/netapp/resource-manager/Microsoft.NetApp/stable/2022-01-01/netapp.json:7167 in [7fb168c](https://github.com/Azure/azure-rest-api-specs/commit/7fb168ce53cf050d217994c22af61eec411eb001). [](commit_id = [7fb168c](https://github.com/Azure/azure-rest-api-specs/commit/7fb168ce53cf050d217994c22af61eec411eb001), deletion_comment = False)

@raosuhas common types `TrackedResource` wont be accepted in autorest validation for a `PATCH `operation.
```
ERROR (PatchBodyParametersSchema/R2016/ARMViolation): Properties of a PATCH request body must not be required. PATCH operation: 'VolumeQuotaRules_Update' Model Definition: 'TrackedResource' Property: 'location'
    - file:///C:/Users/audunn/Source/repos/azure/azure-rest-api-specs/specification/common-types/resource-management/v2/types.json:76:6 ($.definitions.TrackedResource.required)
```

**Comment by audunn** [comment]
@raosuhas changed  `volumeQuotaRule` to reference common types as suggested but now validation complains about missing systemData. How can we fix that ?

```
The response of operation:'VolumeQuotaRules_Get' is defined without 'systemData'. Consider adding the systemData to the response.
Location: [Microsoft.NetApp/stable/2022-01-01/netapp.json#L3243](https://github.com/Azure/azure-rest-api-specs/blob/870c44a321ec5efc8e3e600e63d37fffd7323564/specification/netapp/resource-manager/Microsoft.NetApp/stable/2022-01-01/netapp.json#L3243)
```

File: specification/netapp/resource-manager/Microsoft.NetApp/stable/2021-08-01/netapp.json (hunk 1 above)
Comment (zizw123) [reply specification/netapp/resource-manager/Microsoft.NetApp/stable/2021-08-01/netapp.json:5020]: the common type Resource has common fields such as id, name, type and systemData

Comment (audunn) [reply specification/netapp/resource-manager/Microsoft.NetApp/stable/2021-08-01/netapp.json:5020]: Are you suggesting we change all our tracked resources to reference from common types? 
That does pose a bit of "problem" in validation in the pipeline as `Location` will be flagged for `PATCH` operations in those cases it shares the same type. 
Should we then have a another type for patch operations or is there some other best practice for that scenario?

Comment (audunn) [reply specification/netapp/resource-manager/Microsoft.NetApp/stable/2021-08-01/netapp.json]: @zizw123 @raosuhas  updated PR to reference common types

**Comment by audunn** [comment]
@raosuhas  @zizw123 FYI attempting to reference common types as suggested leads to validation issue where it we get `RequiredReadOnlySystemData` from lintDiff  when netapAccount is referencing systemdata.

am trying to find out why and fix this.

```
    "netAppAccount": {
      "description": "NetApp account resource",
      "type": "object",
      "x-ms-azure-resource": true,
      "required": [
        "location"
      ],
      "allOf": [
        {
          "$ref": "../../../../../common-types/resource-management/v3/types.json#/definitions/TrackedResource"
        }
      ],
      "properties": {
        "properties": {
          "description": "NetApp Account properties",
          "$ref": "#/definitions/accountProperties",
          "x-ms-client-flatten": true
        }
      }
    },
```

**Comment by audunn** [comment]
@zizw123 @raosuhas PR updated. Can you have another look ?

**Comment by audunn** [comment]
> provider should r

updated

**Comment by audunn** [comment]
Note not sure why PowerShell has been added here add is failing in the pipeline (added in a [previous PR](https://github.com/Azure/azure-rest-api-specs/pull/18771) ) 
NetAppFiles Powershell cmdLets are SDK based.

**Comment by jorgecotillo** [comment]
>     "/subscriptions/{subscriptionId}/providers/Microsoft.NetApp/locations/{location}/quotaLimits": {

How are these generated?

---
Refers to: specification/netapp/resource-manager/Microsoft.NetApp/stable/2022-01-01/netapp.json:215 in 1b9383f. [](commit_id = 1b9383fdd3f5173ade5936d325024bb31b4327e5, deletion_comment = False)

**Comment by jorgecotillo** [comment]
>             "_10minutely",

is this valid?

---
Refers to: specification/netapp/resource-manager/Microsoft.NetApp/stable/2022-01-01/netapp.json:5545 in 1b9383f. [](commit_id = 1b9383fdd3f5173ade5936d325024bb31b4327e5, deletion_comment = False)

**Comment by audunn** [comment]
> > ```
> >         "_10minutely",
> > ```
> 
> is this valid?
> 
> Refers to: specification/netapp/resource-manager/Microsoft.NetApp/stable/2022-01-01/netapp.json:5545 in [1b9383f](https://github.com/Azure/azure-rest-api-specs/commit/1b9383fdd3f5173ade5936d325024bb31b4327e5). [](commit_id = [1b9383f](https://github.com/Azure/azure-rest-api-specs/commit/1b9383fdd3f5173ade5936d325024bb31b4327e5), deletion_comment = False)

yes this is valid, was added in api-version 2019-10-01

**Comment by audunn** [comment]
@tianxchen-ms ARMSignedOff can we move this along now ?

**Review by jordan13game — APPROVED** [review]

**Review by rkmanda — APPROVED** [review]
:shipit:

**Pull request merged**